### PR TITLE
Add perl-uri recipe from bioconda

### PR DESCRIPTION
**Pull request by fgiacomoni**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by fgiacomoni** [comment]
@Lain-inrae - Do you see anything on my check issue (win_64)... thank you!

**Comment by fgiacomoni** [comment]
@conda-forge/help-perl recipe has passed all checks and is ready for review. Thank You.

**Comment by fgiacomoni** [comment]
@conda-forge-admin we need a review here

File: recipes/perl-uri/build.sh
Comment (xileF1337) [inline recipes/perl-uri/build.sh:12]: Same here.

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:12]: To ensure that the module version matches your Perl version at runtime, add a weak run export  to the build section like so:

```
build:
  [...]
  run_exports:
    weak:
      - {{ name }} ={{ version }}
```

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:17]: Are you sure that you need C compilers here? I can't see any XS code files in the distribution. Have you tried building without compilers?

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:22]: Please never add explicit Perl version requirements. The Perl version is defined by the global version pinning of CondaForge. The run export should ensure that you get the right Perl version at runtime.

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:40]: Same here: remove Perl version requirement.

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:21]: Okay, you have a lot of dependencies listed here. However, almost all of them are actually core modules and vendored with Perl already. These don't need to be added as requirements. 

A simple way to check for non-core dependencies is a short script like this (you may need to install `cpanm` from [App::Cpanminus](https://metacpan.org/pod/App::cpanminus) and `corelist` from [Module::CoreList](https://metacpan.org/pod/distribution/Module-CoreList/lib/Module/CoreList.pod))

```bash
cpanm --showdeps URI | 
    tail -n+4 | 
    sed 's/~.*$//' |  
    while read m; do 
        corelist "$m" |  grep 'was not in CORE'
    done 
```

Output:

```
perl was not in CORE (or so I think)
Test::Needs was not in CORE (or so I think)
Test::Warnings was not in CORE (or so I think)
Test::Fatal was not in CORE (or so I think)
```

Ignoring the first line, we find only 3 dependencies: `perl-test-needs`, `perl-test-warnings`, and `perl-test-fatal`. You can remove the other ones.

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:59]: Perl also permits distribution under the Perl artistic license. You can use the following keys here:

```
about:
  [...]
  license: GPL-1.0-or-later OR Artistic-1.0-Perl
  license_file:
    - {{ environ["PREFIX"] }}/man/man1/perlartistic.1
    - {{ environ["PREFIX"] }}/man/man1/perlgpl.1
```

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:64]: It is your choice as recipe author, but if you added `conda-forge/perl-packagers` as additional recipe maintainer, we could help to keep this recipe up and running in case of problems ;-)

**Review by xileF1337 — CHANGES_REQUESTED** [review]
Hey there, thanks for your contribution! I have added some remarks to improve your recipe further. After incorporating them, we can merge the PR as far as I am concerned ;-)

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:2]: Please also add:

```
{% set sha256 = "hashValueFromBelow" %}
```

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:10]: Use the Jinja variable defined above here:

```
  sha256: {{ sha256 }}
```

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml:9]: Use the version variable here:
```
  url: https://cpan.metacpan.org/authors/id/O/OA/OALDERS/URI-{{ version }}.tar.gz
```

**Comment by fgiacomoni** [comment]
Dear @xileF1337 Thank you for all yours remarks. I tried to incorporate them in my recipe. Tests are still Ok ;-)

**Comment by fgiacomoni** [comment]
Hi @xileF1337 if you have two minutes to review my work and (fingers crossed) merge my PR ;-)

File: recipes/perl-uri/meta.yaml
Comment (xileF1337) [inline recipes/perl-uri/meta.yaml]: Please also remove the core module dependencies here. Perl and Busines::ISBN should be the only requirements you need. You can also check the current core module list here: https://perldoc.perl.org/modules

**Comment by xileF1337** [comment]
> Hi @xileF1337 if you have two minutes to review my work and (fingers crossed) merge my PR ;-)

I'm sorry, I had a troublesome week and that PR of yours slipped through. Please also remove the core modules from the run requirements and we are ready to merge 8-)

**Comment by fgiacomoni** [comment]
Work (CI Checks) is in progress ! I'm going to ping you asap

**Comment by fgiacomoni** [comment]
Hop - job done - Thank you again

**Review by xileF1337 — APPROVED** [review]

File: recipes/perl-uri/meta.yaml
Comment (fgiacomoni) [reply recipes/perl-uri/meta.yaml:30]: done - sorry for the misunderstanding

**Comment by xileF1337** [comment]
@cbrueffer would you be so kind as to merge this PR? It looks good to me. Thank you!

**Comment by fgiacomoni** [comment]
@cbrueffer any question or issue with the PR ?
Thank you for all.

**Review by cbrueffer — APPROVED** [review]

**Comment by cbrueffer** [comment]
Looks good, thanks!

**Pull request merged**
